builtin/geo3d/mod.rs

```diff
@@ -44,6 +61,32 @@ pub struct Cube {
     pub size_z: Scalar,
 }
 
+
+impl BuiltinModuleDefinition for Cube {
+    fn name() -> &'static str {
+        "cube"
+    }
+
+    fn node(args: &ArgumentMap) -> EvalResult<ObjectNode> {
+        Ok(ObjectNode::new(ObjectNodeInner::Primitive3D(Rc::new(
```

Comment:
I would refactor this to be shorter - like I did in `symbols`.

```suggestion
        Ok(Object::new(ObjectInner::Primitive3D(Rc::new(
```

...and then make `new_` methods - like you did:

```suggestion
        Ok(Object::new_primitive_3d(Object(
```

Let's talk about this.

Reply:
I think I will have to change the ObjectNode tree struct anyway - comes in a next PR ;)